openmp/libomptarget/test/lit.cfg

```diff
@@ -156,8 +156,8 @@ def remove_suffix_if_present(name):
 def add_libraries(source):
     if config.libomptarget_has_libc:
         return source + " " + config.llvm_library_dir + "/libcgpu.a " + \
-               config.library_dir + "/libomptarget.devicertl.a"
-    return source + " " + config.library_dir + "/libomptarget.devicertl.a"
+               config.llvm_library_dir + "/libomptarget.devicertl.a"
+    return source + " " + config.llvm_library_dir + "/libomptarget.devicertl.a"
```

Comment:
But this change can affect both right? In standalone build, I think it is still `config.library_dir` instead of `config.llvm_library_dir`.

Reply:
In the general case, AFAICT both the standalone and LLVM build just default to `./lib` as their output, so it'll succeed superficially. If we want to be sure I could manually set the output to something that's either the LLVM dir or the OpenMP one and then make a new lit variable.

Reply:
With that being said, I'm not sure what this patch tries to fix or improve. Does it try to set the output path at build time or install time?

Reply:
It's build time. The issue is that I wanted `libc` to fish this out of the install directory, but where it's written to is variable and in a random directory. I wanted it to be next to `libomptarget.so` so I have a single static search path I can use for internal resolution of the library.

Reply:
I tried your patch and I think it will indeed break standalone build.
```
config.library_dir = "/gpfs/jlse-fs0/users/ac.shilei.tian/build/openmp/release/libomptarget"
config.llvm_library_dir = "/gpfs/jlse-fs0/users/ac.shilei.tian/build/llvm/release/./lib"
```
The DeviceRTL library is in `/gpfs/jlse-fs0/users/ac.shilei.tian/build/openmp/release/libomptarget`.

Reply:
Guess I'll set the `library_dir` more correctly. Thanks.

Reply:
And even w/o this patch, `libomptarget.devicertl.a` sits next to `libomptarget.so`.
```
➜  libomptarget ll
total 4.1M
drwxr-xr-x  8 ac.shilei.tian jlse 4.0K Nov 28 18:20 ./
drwxr-xr-x  8 ac.shilei.tian jlse 4.0K Nov 28 18:19 ../
drwxr-xr-x  2 ac.shilei.tian jlse 4.0K Nov 28 18:19 CMakeFiles/
-rw-r--r--  1 ac.shilei.tian jlse 2.4K Nov 28 18:19 cmake_install.cmake
drwxr-xr-x  3 ac.shilei.tian jlse 256K Nov 28 18:20 DeviceRTL/
-rw-r--r--  1 ac.shilei.tian jlse 1.8M Nov 28 18:20 libomptarget.devicertl.a
lrwxrwxrwx  1 ac.shilei.tian jlse   32 Nov 28 18:20 libomptarget.rtl.amdgpu.so -> libomptarget.rtl.amdgpu.so.18git*
-rwxr-xr-x  1 ac.shilei.tian jlse 443K Nov 28 18:20 libomptarget.rtl.amdgpu.so.18git*
lrwxrwxrwx  1 ac.shilei.tian jlse   30 Nov 28 18:20 libomptarget.rtl.cuda.so -> libomptarget.rtl.cuda.so.18git*
-rwxr-xr-x  1 ac.shilei.tian jlse 360K Nov 28 18:20 libomptarget.rtl.cuda.so.18git*
lrwxrwxrwx  1 ac.shilei.tian jlse   32 Nov 28 18:20 libomptarget.rtl.x86_64.so -> libomptarget.rtl.x86_64.so.18git*
-rwxr-xr-x  1 ac.shilei.tian jlse 303K Nov 28 18:20 libomptarget.rtl.x86_64.so.18git*
lrwxrwxrwx  1 ac.shilei.tian jlse   21 Nov 28 18:20 libomptarget.so -> libomptarget.so.18git*
-rwxr-xr-x  1 ac.shilei.tian jlse 209K Nov 28 18:20 libomptarget.so.18git*
drwxr-xr-x 11 ac.shilei.tian jlse 4.0K Nov 28 18:19 plugins-nextgen/
drwxr-xr-x  3 ac.shilei.tian jlse 4.0K Nov 28 18:19 src/
drwxr-xr-x  8 ac.shilei.tian jlse 4.0K Nov 28 18:19 test/
drwxr-xr-x  5 ac.shilei.tian jlse 4.0K Nov 28 18:19 tools/
```

Reply:
Definitely doesn't in my build tree, both projects and runtimes. I think standalone defaults to `./lib` as well?

Reply:
It is not in `./lib` because it is build time. Eventually after installation it will be in `./lib`, but it is in `config.library_dir`.

Reply:
Ah, the build is what I care about. It currently gets installed to the same place, yes. I'm fishing these libraries out of the user's build so it needs to be in a consistent place.

Reply:
Have you tried  ninja -t missingdeps?